ingestion/src/metadata/ingestion/source/dashboard/mstr/client.py

```diff
@@ -45,32 +46,92 @@ class MSTRClient:
 
     def _get_base_url(self, path=None):
         if not path:
-            return f"{clean_uri(self.config.hostPort)}/{API_VERSION}/"
+            return f"{clean_uri(self.config.hostPort)}/{API_VERSION}"
         return f"{clean_uri(self.config.hostPort)}/{API_VERSION}/{path}"
 
-    def _get_mstr_session(self) -> MSTRRESTSession:
-        try:
-            session = MSTRRESTSession(base_url=self._get_base_url())
-            session.login(
-                username=self.config.username,
-                password=self.config.password.get_secret_value(),
-            )
-            return session
-
-        except KeyError as exe:
-            msg = "Failed to fetch mstr session, please validate credentials"
-            raise SourceConnectionException(msg) from exe
-
-        except Exception as exc:
-            msg = f"Unknown error in connection: {exc}."
-            raise SourceConnectionException(msg) from exc
-
     def __init__(
         self,
         config: MstrConnection,
     ):
         self.config = config
-        self.session = self._get_mstr_session()
+
+        self.auth_params: AuthHeaderCookie = self._get_auth_header_and_cookies()
+
+        client_config = ClientConfig(
+            base_url=clean_uri(config.hostPort),
+            api_version=API_VERSION,
+            extra_headers=self.auth_params.auth_header,
+            allow_redirects=True,
+            cookies=self.auth_params.auth_cookies,
+        )
+
+        self.client = REST(client_config)
+        self._set_api_session()
+
+    def _get_auth_header_and_cookies(self) -> Optional[AuthHeaderCookie]:
+        """
+        Send a request to authenticate the user and get headers and
+        """
+        try:
+            data = {
+                "username": self.config.username,
+                "password": self.config.password.get_secret_value(),
+                "loginMode": 8,
+                "applicationType": 35,
```

Comment:
we should document in the code itself the magic numbers (or better add a meaningful constant).

also if there any doc references from mstr we can add them here too

Reply:
Let's put constants, e.g., `LOGIN_MODE_POTATO = 8`

Reply:
Added the constant